pkg/internal/controller/controller.go

```diff
@@ -175,7 +175,7 @@ func (c *Controller[request]) Start(ctx context.Context) error {
 		// caches.
 		errGroup := &errgroup.Group{}
 		for _, watch := range c.startWatches {
-			log := c.LogConstructor(nil).WithValues("source", fmt.Sprintf("%s", watch))
+			log := c.LogConstructor(nil).WithValues("source", watch.String())
```

Comment:
It is an interesting issue - How about  rather then requiring the implementation of `fmt.Stringer`, we check if it is implemented and if yes use that, otherwise use the Type (i.E. `fmt.Sprintf("%T", watch)`)? That way we avoid the breaking change